### PR TITLE
feat: add vault.approle options

### PR DESCRIPTION
**Pull request by rasta-rocket**
* allow service to login against a custom approle path in vault
* update documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -81,6 +81,7 @@ To deploy it just run `kubectl apply -f secretdefinition-sample.yaml`
 | `vault.role-id` | `""` | Vault appRole `role_id`. `VAULT_ROLE_ID` environment would take precedence. |
 | `vault.secret-id` | `""` | Vault appRole `secret_id`. `VAULT_SECRET_ID` environment would take precedence. |
 | `vault.engine` | kv2 | Vault secrets engine to use. Only key/value engines supported. Default is kv version 2 |
+| `vault.approle` | approle | Vault approle path |
```
Comment (fcgravalos) [inline README.md]: I think `vault.approle-path` would be more consistent. What do you think?

File: backend/vault.go [hunk 2]
```diff
@@ -33,7 +35,8 @@ func (c *client) vaultLogin() error {
 		"role_id":   c.roleID,
 		"secret_id": c.secretID,
 	}
-	resp, err := c.logical.Write("auth/approle/login", appRole)
+	approleLoginPath := fmt.Sprintf("auth/%s/login", c.approlePath)
+	resp, err := c.logical.Write(approleLoginPath, appRole)
```
Comment (fcgravalos) [inline backend/vault.go]: I think we can squash these 2 lines in just 1, wdyt?

File: README.md (hunk 1 above)
Comment (rasta-rocket) [reply README.md]: > I think `vault.approle-path` would be more consistent. What do you think?

yep

File: backend/vault.go (hunk 2 above)
Comment (rasta-rocket) [reply backend/vault.go]: > I think we can squash these 2 lines in just 1, wdyt?

ok 👍

**Pull request merged**